### PR TITLE
@ #58 | should install require modules when ansible was installed

### PR DESCRIPTION
**Pull request by hieptranquoc**
connect #58 
related: https://github.com/teracyhq-incubator/teracy-dev-entry-k8s/issues/14
please check this PR! Thank you!

### CODE REVIEW

File: provisioners/ansible/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+k8s-requirements.retry
```
Comment (hoatle) [inline provisioners/ansible/.gitignore]: *.retry

File: provisioners/ansible/k8s-requirements.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+---
+- hosts: all
+  become: yes
+  any_errors_fatal: "{{ any_errors_fatal | default(true) }}"
+  tasks:
+    - name: check pip exists
+      command: which pip
+      register: pip_check
```
Comment (hoatle) [inline provisioners/ansible/k8s-requirements.yml]: pip_exist

File: provisioners/ansible/k8s-requirements.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
+---
+- hosts: all
+  become: yes
+  any_errors_fatal: "{{ any_errors_fatal | default(true) }}"
+  tasks:
+    - name: check pip exists
```
Comment (hoatle) [inline provisioners/ansible/k8s-requirements.yml]: - name: Check pip exists

File: provisioners/ansible/k8s-requirements.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        - stat_pip_result.stat.exists == True
+
+    - debug:
+        msg: "pip has been installed"
+      when: (pip_install is succeeded) or (pip_check is succeeded)
+
+    - stat: path=/vagrant/{{ kubespray_lookup_path }}/kubespray/requirements.txt
+      register: requirement_file
+
+    - debug:
```
Comment (hoatle) [inline provisioners/ansible/k8s-requirements.yml]: should send warning

File: provisioners/ansible/k8s-requirements.yml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        msg: "/vagrant/{{ kubespray_lookup_path }}/kubespray/requirements.txt not exists"
+      when: requirement_file.stat.exists == False
+
+    - name: install teracy-dev-k8s requirements
```
Comment (hoatle) [inline provisioners/ansible/k8s-requirements.yml]: Install Kuberspray pip requirements

File: lib/teracy-dev-k8s/processors/settings.rb [hunk 6]
```diff
@@ -157,6 +157,25 @@ def generate_nodes(k8s_config)
           if i == num_instances
 
             if k8s_config['ansible']['mode'] == 'guest'
+
+              k8s_requirement_extra_vars = {
+                "kubespray_lookup_path" => kubespray_lookup_path
+              }
+
+              provisioner_k8s_requirement = {
+                "_id" => "requirement-0",
```
Comment (hoatle) [inline lib/teracy-dev-k8s/processors/settings.rb]: "_id" => "k8s-pip-requirements"

**Review by hoatle — CHANGES_REQUESTED** [review]
@hieptranquoc pls check my comments

**Comment by hieptranquoc** [comment]
I've updated! Please check this PR again! Thank you!

**Review by hoatle — APPROVED** [review]
LGTM

**Pull request merged**